Fix some search issues

jrainville opened the pull request:
### What does the PR do

Fixes #10184

- Fixes the order of channels in the location menu
- Makes sure that only channels that can be read are shown in the location menu

Missing part: the emoji size wasn't fixed as I need help

### Affected areas

Location menu in the search popup

### Architecture compliance

- [x] I am familiar with the application architecture and agreed good practices.
My PR is consistent with this document: [Status Desktop Architecture Guide](https://github.com/status-im/status-desktop/blob/master/CONTRIBUTING.md)

### Screenshot of functionality (including design for comparison)

Fixed order: 
![image](https://github.com/user-attachments/assets/1d6be51d-cda1-4fb3-82de-110a8cf84bb6)

Spectated community not appearing:
![image](https://github.com/user-attachments/assets/2592ae73-324f-4b66-a787-a9aeead569f7)

Encrypted channels don't show:
![image](https://github.com/user-attachments/assets/00979634-2ef2-431a-afe5-32c1088b6104)

### Impact on end user

Fixes the location menu

### How to test

1. Be part of a community with lots of channels with categories
2. Press CTRL+F
3. Check the location menu for the channels of that community
4. Have an encrypted channel (a channel the user can't view)
5. As a member of the channel (not owner), check the menu again
6. The encrypted channel doesn't show

### Risk 

Tick **one**:
- [x] Low risk: 2 devs MUST perform testing as specified above and attach their results as comments to this PR **before** merging.
- [ ] High risk: QA team MUST perform additional testing in the specified affected areas **before** merging.

Worst case the order of channel could still be messed up

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | b187c0cd | [#1](https://ci.status.im/job/status-desktop/job/prs/job/macos/job/aarch64/job/package/job/PR-16943/1/) | 2024-12-11 21:35:56 | ~6 min | `macos/aarch64` | [:apple:`dmg`](https://status-im-desktop-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Desktop-241211-212925-b187c0-pr16943-aarch64.dmg) |

caybro commented:
Emoji rendering fixed in the 092d35723d1f404ae91d3aef3136f9db4cded5d0 commit

caybro commented:
"Chat" no longer has submenus (this PR vs. master):

![image](https://github.com/user-attachments/assets/0962bee3-b4cd-43ff-bd57-73f942faaf06)

![image](https://github.com/user-attachments/assets/d6739bfd-aac8-4636-ac56-76dfcc45c0b2)

Another problem, in both branches, the list of Status CC channels is very incomplete

caybro reviewed (changes requested):
Nice work, but need to fix the "Chat" submenu items regression, and preferably also the (incomplete) list of Status CC channels

jrainville commented:
> Nice work, but need to fix the "Chat" submenu items regression, and preferably also the (incomplete) list of Status CC channels

@caybro nice finds. I fixed in my second commit. You'll need to reset the branch

caybro reviewed (approved):
Looks good!

I just amended my last emoji fix commit with a little fix for group chats

jrainville commented:
@micieslak @osmaczko friendly reminder to review

micieslak reviewed (approved):
lgtm!

jrainville commented:
@alexjba you've been randomly chosen to test this
> List Randomizer
There were 5 items in your list. Here they are in random order:
> 
> alexjba
micieslak
caybro
osmaczko
alaibe

alexjba reviewed (approved):
LGTM! I have two findings below, but looks to me like it's not related to this PR. Please let me know what you think of these:

1. One community icon doesn't show in the search context menu
https://status.app/c/GzwAAGTGNrOXuOplX3Esx8vOUxjSqDQOVVEcttR_-H5JaYfvX85J_TjqcfXD_qnD37kypONGHKV-3ZiQiwI=#zQ3shnortLaN5azfhAUYM6Abmqyjp9NP3Wq3nH38DibS6Xzqz

<img width="1201" alt="Screenshot 2024-12-19 at 11 55 32" src="https://github.com/user-attachments/assets/59872960-b50b-4920-8b5e-ab82728e5a13" />


2. Chats sorting order is different in the context menu

<img width="1196" alt="Screenshot 2024-12-19 at 11 56 28" src="https://github.com/user-attachments/assets/ad44b5f9-a33a-49e7-acf6-699fa631f3ee" />

jrainville commented:
> One community icon doesn't show in the search context menu

Weird. I would expect them to either all work or all fail. It seems like it also doesn't have chats. I'll check if I can reproduce, but I'm not sure if I caused that.

> Chats sorting order is different in the context menu

Ah, the issue was only about the community order. I'll check real quick if I can get the same order for the personal chats. I think it uses the last message timestamp, so maybe it's harder.

jrainville commented:
@alexjba I fixed the ordering of the personal chat (see last commit).

I also found a small bug with the color of the channels with emojis. I fixed that too.

As for the missing icon on the community, I couldn't reproduce. Can you check if you can see it on master too? I haven't touched the section items, only the subitems (chats), so I don't know how I could have done that.

alexjba commented:
> As for the missing icon on the community, I couldn't reproduce. Can you check if you can see it on master too? I haven't touched the section items, only the subitems (chats), so I don't know how I could have done that.

Looks good on master. I can have a look tomorrow if you can't reproduce

<img width="1199" alt="Screenshot 2024-12-19 at 22 25 23" src="https://github.com/user-attachments/assets/336fb251-a1cb-451a-ae4a-d338edd63f3b" />

alexjba commented:
It's probably due to the condition here https://github.com/status-im/status-desktop/pull/16943/files#diff-503ac13f5e21b6a8dc452952c801a2239dd091d538cc9e092ca0cc90bd3b31fcR74

Found a way to reproduce:
Create a community with `Request to join required`. Leave the community and restart the app.

alexjba commented:
But on top of this, I see a new regression on the group chats icons

<img width="1199" alt="Screenshot 2024-12-19 at 23 45 21" src="https://github.com/user-attachments/assets/afa05710-1446-4cc7-a4c8-f6278297ec64" />

jrainville commented:
Nice finds @alexjba 

I fixed the group chat issue by adding a new property called `isImage`. We didn't have a good way before to know if the image was an emoji or an image, so we either broke one or the other. Now both work.

As for the community issue, I just make it so that you can only search communities you actually joined. This in line with the original ticket too.
